### ISSUE TITLE
"Type aliases" like `X{T} = Y{T, S}` are not handled

### ISSUE THREAD

**Issue by tkf** [JuliaStaging/JuliaVariables.jl#11]
It seems that "type aliases" are not handled correctly at the moment:

```julia
julia> solve(:(X{T} = Y{T, S}))
:((@global X){@global T} = (@global Y){@global T, @global S})
```

I suppose the result should be

```julia
:((@global X){@T} = (@global Y){@T, @global S})
```

?

**Comment by thautwarm** [comment]
Yes, it's a bit complex, and requires more efforts to match the correct semantics. 
I remembered you've sent the link of the scheme file(julia-syntax.scm) which implements the syntax analysis, but it's in quite a huge scale so I cannot reach the same functions immediately.

**Comment by tkf** [comment]
Yes, Julia language is a huge target and I admire your courage to implement JuliaVariables.jl :)

> the scheme file(julia-syntax.scm) which implements the syntax analysis

I'm not familiar with compiler/language things, but I thought this analysis is done while lowering.  But I'm not sure.
